tests/python/direct/test_sdpa.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# Owner(s): ["module: nvfuser"]
+
+import itertools
+import math
+import pytest
+import torch
+import torch.nn.functional as F
+from functools import partial
+from nvfuser_direct import (
+    FusionDefinition,
+    DataType,
+)
+from python.direct_utils import is_pre_ampere
+
+
+@pytest.mark.skipif(
+    is_pre_ampere(),
+    reason="Flash Attention is only supported on Ampere and newer devices.",
+)
+def test_softmax_logsumexp(nvfuser_direct_test):
+    def fusion_func(fd: FusionDefinition) -> None:
+        q = fd.define_tensor(
+            shape=[-1, -1, -1, -1],
+            dtype=DataType.BFloat16,
+        )
+        k = fd.define_tensor(
+            shape=[-1, -1, -1, -1],
+            dtype=DataType.BFloat16,
+        )
+        v = fd.define_tensor(
+            shape=[-1, -1, -1, -1],
+            dtype=DataType.BFloat16,
+        )
+        (
+            _,
+            lse,
+            *_,
+        ) = fd.ops.sdpfa_fwd(q, k, v, dropout_p=None, is_causal=None, scale=None)
+        fd.add_output(lse)
+
+    n, h, l, s, e = 1, 1, 4, 4, 2
+    inputs = [
+        torch.ones((n, h, l, e), dtype=torch.bfloat16, device="cuda"),
+        torch.ones((n, h, s, e), dtype=torch.bfloat16, device="cuda"),
+        torch.ones((n, h, s, e), dtype=torch.bfloat16, device="cuda"),
+    ]
+
+    from torch.nn.attention import SDPBackend, sdpa_kernel
+
+    with sdpa_kernel(SDPBackend.FLASH_ATTENTION):
+        nvf_out, _ = nvfuser_direct_test.exec_nvfuser(
+            fusion_func,
+            inputs,
+        )
+    # Ignoring size-1 dimensions, `q @ k^T / sqrt(e)` generates a `l`x`s`
+    # matrix full of `sqrt(e)`s.  Therefore, the logsumexp of each row is
+    # expected to be log(exp(sqrt(e)) * s) = log(s) + sqrt(e).
+    torch.testing.assert_close(
+        nvf_out[0].cpu(), torch.full((n, h, l), math.log(s) + e**0.5)
```

Comment:
For testing the `logsumexp` output, shouldn't you be testing the second output in the tuple?

```
A tuple of (output, log_sumexp, philox_seed, philox_offset).
      )",
```

Reply:
The fusion returns only the logsumexp as an output.
```python
            (
                _,
                lse,
                *_,
            ) = fd.ops.sdpfa_fwd(q, k, v, dropout_p=None, is_causal=None, scale=None)
            fd.add_output(lse)
```